spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -250,13 +248,13 @@ def _measure_volume(self, im_data, p_lst, idx):
         printv('  Volume : ' + str(np.round(vol_tot_cur, 2)) + ' mm^3', self.verbose, type='info')
 
     def _measure_length(self, im_data, p_lst, idx):
-        length_cur = np.sum([np.cos(self.angles[zz]) * p_lst[2] for zz in np.unique(np.where(im_data)[2])])
+        length_cur = np.sum([p_lst[2] / np.cos(self.angles[zz]) for zz in np.unique(np.where(im_data)[2])])
         self.measure_pd.loc[idx, 'length [mm]'] = length_cur
         printv('  (S-I) length : ' + str(np.round(length_cur, 2)) + ' mm', self.verbose, type='info')
 
     def _measure_diameter(self, im_data, p_lst, idx):
         area_lst = [np.sum(im_data[:, :, zz]) * np.cos(self.angles[zz]) * p_lst[0] * p_lst[1] for zz in range(im_data.shape[2])]
```

Comment:
I noticed you changed `_measure_length` from `np.cos(self.angles[zz]) * p_lst[2]` to `p_lst[2] / np.cos(self.angles[zz])`.

Is there any chance the same needs to be done for the `_measure_diameter` instance of `np.cos`, too?